utils/fill_in_results.sh

```diff
@@ -0,0 +1,147 @@
+    insights-run --help > /dev/null 2>&1 || { echo >&2 -e "${RED}Insights OCP needs to be installed and initialized. Aborting.${NC}"; exit 1; }
+    echo -e "${GREEN}Insights OCP seems to be initialized properly${NC}"
+}
+
+
+# Decompress input file taken from Kraken or any other external source
+function decompress_input() {
+    echo -e "${BLUE}Decompressing input file $1${NC}"
+    tar xvfz "$1"
+    echo -e "${GREEN}Done${NC}"
+}
+
+
+# Input tarball has very strange directory structure (ask other teams why on Earth do they need it)
+# We just move all files from this structure into one "flat" workdir
+function move_input_to_workdir() {
+    find . -name "*.tar.gz" -exec mv -i -- {} "$1" \;
+}
+
+
+# Remove now empty temporary directories
+function cleanup_temp_directories() {
+    # shellcheck disable=SC2045,SC2035
+    for temp_directory in $(ls -1 -d */)
+    do
+        if [ "$temp_directory" != "$1/" ]
+        then
+            echo "Removing ${temp_directory}"
+            rm -rf "${temp_directory}"
+        else
+            echo -e "${BLUE}Skipping ${temp_directory}${NC}"
+        fi
+    done
+}
+
+# Apply all OCP rules for all input files (gzipped tarballs)
+# Output is formatted to be human readable
+function apply_ocp_rules() {
+    cwd=$(pwd)
+    cd "$1" || exit
+
+    echo -e "${BLUE}Starting applying OCP rules${NC}"
+    for input in *.tar.gz
+    do
+        echo "$input"
+        insights-run -p ccx_rules_ocp -f json "${input}" > x.json
+        python -m json.tool < x.json > "$input.json"
+    done
+    echo -e "${GREEN}Done${NC}"
+
+    # leftover
+    rm x.json
+
+    cd "${cwd}" || exit
+}
+
+
+# Clean tarballs after OCP rules are applied
+function clean_tarballs() {
+    rm "$1"/*.tar.gz
+}
+
+
+# Anonymize all OCP rules results
+function anonymize() {
+    cwd=$(pwd)
+    cd "$1" || exit
+
+    echo -e "${BLUE}Anonymizing OCP rules results${NC}"
+    python3 "${cwd}"/anonymize.py
+    echo -e "${GREEN}Done${NC}"
+
+    cd "${cwd}" || exit
+}
+
+
+# Convert OCP rules results into JSONs compatible with aggregator
+function convert_to_report() {
+    cwd=$(pwd)
+    cd "$1" || exit
+
+    echo -e "${BLUE}Converting OCP rules results into JSONs compatible with aggregator${NC}"
+    python3 "${cwd}"/2report.py "$2" "$3"
+    echo -e "${GREEN}Done${NC}"
+
+    cd "${cwd}" || exit
+}
+
+
+# Perform cleanup after all previous operations
+function cleanup_workdir() {
+    echo -e "${BLUE}Cleanup${NC}"
+
+    rm -f "$1"/*.tar.gz.json
+    rm -f "$1"/s_*.json
+
+    echo -e "${GREEN}Done${NC}"
+}
+
+
+# check parameters first
+if [ $# -lt 3 ]
+then
+    echo -e "${BLUE}Usage:${NC}"
+    echo "    fill_in_results.sh archive.tar.bz org_id cluster_name"
+    echo -e "${BLUE}Example:${NC}"
+    echo "    fill_in_results.sh external-rules-archives-2020-03-31.tar 11789772 5d5892d3-1f74-4ccf-91af-548dfc9767aa"
+    exit 1
+fi
+
+check_insights_availability
+decompress_input exernal-rules-archives-2020-03-31.tar
```

Comment:
LGTM so far, I assume this will be changed to the param?

Reply:
definitely, yes, $1